texar/tf/module_base.py

```diff
@@ -68,7 +70,16 @@ class ModuleBase(object):
     """
 
     def __init__(self, hparams=None):
-        self._hparams = HParams(hparams, self.default_hparams())
+        if not hasattr(self, '_hparams'):
+            self._hparams = HParams(hparams, self.default_hparams())
+        else:
+            # Probably already parsed by subclasses. We rely on subclass
+            # implementations to get this right.
+            # As a sanity check, we require `hparams` to be `None` in this case.
+            if hparams is not None:
+                raise ValueError(
+                    "`self._hparams` is already assigned, but `hparams` "
+                    "argument is not None.")
```

Comment:
Error msg polished a bit:

`self._hparams` already exists. Argument `hparams` must be set to `None` in this case.